src/base/undistortion.cc

```diff
@@ -531,6 +532,183 @@ void CMPMVSUndistorter::Undistort(const size_t reg_image_idx) const {
+  }
+
+  GetTimer().PrintMinutes();
+}
+
+// For format details see:
+// https://github.com/simonfuhrmann/mve/blob/master/libs/mve/bundle_io.cc
+void MVEUndistorter::WriteBundle() const {
+  const std::string bundle_path = JoinPaths(output_path_, "synth_0.out");
+  std::ofstream file(bundle_path, std::ios::trunc);
+  CHECK(file.is_open()) << bundle_path;
+  file << "drews 1.0" << std::endl;
+  const auto num_images = reconstruction_.Images().size();
+  file << num_images << " "
+       << reconstruction_.NumPoints3D() << std::endl;
+  Eigen::IOFormat rFmt(Eigen::StreamPrecision, Eigen::DontAlignCols, " ",
+                       "\n", "", "", "", "");
+  Eigen::IOFormat tFmt(Eigen::StreamPrecision, Eigen::DontAlignCols, " ",
+                       " ", "", "", "", "");
+  int count = 0;
+  std::unordered_map<image_t, int> dense_image_ids;
+  for (const auto& image_id: reconstruction_.RegImageIds()) {
+    const auto& image = reconstruction_.Image(image_id);
+    const auto& camera = reconstruction_.Camera(image.CameraId());
+    // Write out 0 radial distortion as we expect MVE to use undistorted images
+    file << camera.NormalizedFocalLength() << " 0 0" << std::endl;
+    file << Eigen::Quaterniond(image.Qvec())
+                .normalized()
+                .toRotationMatrix()
+                .format(rFmt) << std::endl;
+    file << image.Tvec().format(tFmt) << std::endl;
+    dense_image_ids[image_id] = count++;
+  }
+
+  for (const auto& point3d_id: reconstruction_.Point3DIds()) {
+    const auto& point3d = reconstruction_.Point3D(point3d_id);
+    file << point3d.X() << " " << point3d.Y() << " " << point3d.Z()
+         << std::endl;
+    file << (int)point3d.Color(0) << " " << (int)point3d.Color(1)
+         << " " << (int)point3d.Color(2) << std::endl;
+    const auto& track = point3d.Track();
+    file << track.Length() << " ";
+    std::unordered_map<image_t, std::vector<point2D_t>> image_points;
+    for (const auto& element: track.Elements()) {
+      image_points[element.image_id].push_back(element.point2D_idx);
+    }
+    for (const auto& item : image_points) {
+      const image_t image_id = item.first;
+      // const auto& image = reconstruction_.Image(image_id);
+      CHECK_GT(item.second.size(), 0);
+      point2D_t point2d_idx = item.second[0];
+      if (item.second.size() > 0) {
+        point2d_idx = ChooseLowestReprojectionErrorPoint(image_id, point3d_id,
+                                                         item.second);
+      }
+      CHECK(dense_image_ids.find(image_id) != dense_image_ids.end());
+      // const auto& point2d = image.Point2D(point2d_idx);
+      file << dense_image_ids[image_id] << " " << point2d_idx << " 0 ";
+      // Photosynth format does not need the image feature coordinates?
+      // << point2d.X() << " " << point2d.Y();
+    }
+    file << std::endl;
+  }
+}
+
+point2D_t MVEUndistorter::ChooseLowestReprojectionErrorPoint(
+    image_t image_id, point3D_t point3d_id,
```

Comment:
Make these arguments const please.

Reply:
Fix these please and I will merge. Thanks!

Reply:
Thats not the usual c++ style (I follow google coding style guide) to make pass by value args const. I can do it if you insist for codebase consistency reason.